src/Commands/Connection/SetCommand.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Connection;
+
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Terminus\Collections\Sites;
+use Terminus\Exceptions\TerminusException;
+use Terminus\Models\Environment;
+
+/**
+ * Class SetCommand
+ * @package Pantheon\Terminus\Commands\Connection
+ */
+class SetCommand extends TerminusCommand
+{
+    /**
+     * Set git or sftp connection mode on a site's dev or multidev environment
+     *
+     * @authorized
+     *
+     * @command connection:set
+     *
+     * @param string $environment Name of the environment to set. Note that you cannot change 'test' or 'live'.
+     * @param string $mode        Connection mode, one of: <git|sftp>
+     *
+     * @return bool
+     *
+     */
+    public function connectionSet($environment, $mode)
+    {
+        $site_env = explode('.', $environment);
+        $sites = new Sites();
+        $site = $sites->get($site_env[0]);
+        $env   = $site->environments->get($site_env[1]);
```

Comment:
Well... unless the input was just `site` which is missing the `.env` portion. I felt this approach was safer and I had the appropriate guards (and tests) in place until I was informed that the argument validation is coming and I don't need to be that defensive. So... yes it can, and should be simplified now. Thanks!